src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -8,6 +8,29 @@ namespace System.IO
     /// <summary>Contains internal path helpers that are shared between many projects.</summary>
     internal static partial class PathInternal
     {
+        internal const string LongPathPrefix = @"\\?\";
+        internal const string UNCPathPrefix = @"\\";
+        internal const string UNCLongPathPrefixToInsert = @"?\UNC\";
+        internal const string UNCLongPathPrefix = @"\\?\UNC\";
+
+        internal static readonly char[] InvalidPathChars =
+        {
+            '\"', '<', '>', '|', '\0',
+            (char)1, (char)2, (char)3, (char)4, (char)5, (char)6, (char)7, (char)8, (char)9, (char)10,
+            (char)11, (char)12, (char)13, (char)14, (char)15, (char)16, (char)17, (char)18, (char)19, (char)20,
+            (char)21, (char)22, (char)23, (char)24, (char)25, (char)26, (char)27, (char)28, (char)29, (char)30,
+            (char)31
+        };
+
+        internal static readonly char[] InvalidPathCharsWithAdditionalChecks = // This is used by HasIllegalCharacters
+        {
+            '\"', '<', '>', '|', '\0',
+            (char)1, (char)2, (char)3, (char)4, (char)5, (char)6, (char)7, (char)8, (char)9, (char)10,
+            (char)11, (char)12, (char)13, (char)14, (char)15, (char)16, (char)17, (char)18, (char)19, (char)20,
+            (char)21, (char)22, (char)23, (char)24, (char)25, (char)26, (char)27, (char)28, (char)29, (char)30,
+            (char)31, '*', '?'
+        };
+
```

Comment:
Is there a way we can avoid defining all of these things twice, once in Path and once in PathInternal?  For example, would it make sense to have these in a shared location, with a #if used to change the name of the type they're included in?  I realize we already have some duplication, but I was unhappy with that to begin with, and this just makes it worse.  I'm ok with going ahead with this to unblock things, but I'd love to see a subsequent solution imposed.